c++/src/kj/compat/tls.c++

```diff
@@ -405,31 +416,108 @@ private:
 // =======================================================================================
 // Implementations of ConnectionReceiver, NetworkAddress, and Network as wrappers adding TLS.
 
-class TlsConnectionReceiver final: public kj::ConnectionReceiver {
+template <typename T>
+class InputQueue {
+  // InputQueue is an async FIFO queue that has many sources but only one sink.
+  // TODO(pr) Should I find a header home for this somewhere?
```

Comment:
One nice thing about putting it in a header would be that we can unit test it directly.

But the implementation does appear to be coupled to the specific use case in a few ways. If we put it in a header we might want to generalize to support multiple waiters, and maybe remove the `push(promise)` method (make `pushRaw()` be the normal `push()`). I dunno.

Reply:
I think the shift to doubly-linked list is enough of an increase in complexity that I want to write separate tests. Header it is!